exchanges/request/request.go

```diff
@@ -19,14 +19,25 @@ import (
 )
 
 // New returns a new Requester
-func New(name string, httpRequester *http.Client, l Limiter) *Requester {
-	return &Requester{
-		HTTPClient:           httpRequester,
-		Limiter:              l,
-		Name:                 name,
-		timeoutRetryAttempts: TimeoutRetryAttempts,
-		timedLock:            timedmutex.NewTimedMutex(DefaultMutexLockTimeout),
+func New(name string, httpRequester *http.Client, opts ...RequesterOption) *Requester {
+	maxRetries := MaxRetryAttempts
+	if maxRetries == 0 && TimeoutRetryAttempts > 0 {
```

Comment:
'Max retries' replaces 'timeout retry attempts', but aiming to keep some backward compatibility to avoid breaking existing configs. 'Timeout retries' isn't accurate anymore, as we retry on more than just timeouts now. As it stands, this doesn't work as `MaxRetryAttempts` is always non-zero, so `TimeoutRetryAttempts` will never be considered.

Need feedback regarding what the preference is for how this behaves / whether the existing 'timeout retry attempts' flag should be dropped entirely

Reply:
Originally we wanted to differentiate between network transient errors and outright errors, say a bad request to an invalid endpoint or an internal server error. Because we have a connection monitor now, I think it's fine to ditch the timeout retry attempts var (as this is a cmdline param, it's OK to get rid of. Anything config related however, requires an upgrade path)